tests/src/test/scala/common/rest/WskRest.scala

```diff
@@ -0,0 +1,1482 @@
+
+  /**
+   * Gets entity from collection.
+   *
+   * @param name either a fully qualified name or a simple entity name
+   * @param expectedExitCode (optional) the expected exit code for the command
+   * if the code is anything but DONTCARE_EXIT, assert the code is as expected
+   */
+  override def get(name: String,
+                   expectedExitCode: Int = OK.intValue,
+                   summary: Boolean = false,
+                   fieldFilter: Option[String] = None,
+                   url: Option[Boolean] = None)(implicit wp: WskProps): RestResult = {
+    val (ns, entity) = getNamespaceActionName(name)
+    val entPath = Path(s"$basePath/namespaces/$ns/$noun/$entity")
+    val resp = getWhiskEntity(entPath)(wp).futureValue
+    val r = new RestResult(resp.status, getRespData(resp))
+    if (expectedExitCode != DONTCARE_EXIT)
+      if (r.statusCode.intValue != expectedExitCode)
+        r.statusCode.intValue shouldBe expectedExitCode
+    r
+  }
+}
+
+trait DeleteFromCollectionRest extends BaseDeleteFromCollection {
+  self: RunWskRestCmd =>
+
+  /**
+   * Deletes entity from collection.
+   *
+   * @param name either a fully qualified name or a simple entity name
+   * @param expectedExitCode (optional) the expected exit code for the command
+   * if the code is anything but DONTCARE_EXIT, assert the code is as expected
+   */
+  override def delete(entity: String, expectedExitCode: Int = OK.intValue)(implicit wp: WskProps): RestResult = {
+    val r = deleteEntity(entity)(wp)
+    if (expectedExitCode != DONTCARE_EXIT)
+      if (r.statusCode.intValue != expectedExitCode)
+        r.statusCode.intValue shouldBe expectedExitCode
+    r
+  }
+
+  def deleteEntity(name: String)(implicit wp: WskProps): RestResult = {
+    val (ns, entity) = getNamespaceActionName(name)
+    val path = Path(s"$basePath/namespaces/$ns/$noun/$entity")
+    val resp = deleteEntity(path)(wp)
+    new RestResult(resp.status, getRespData(resp))
+  }
+
+  /**
+   * Deletes entity from collection but does not assert that the command succeeds.
+   * Use this if deleting an entity that may not exist and it is OK if it does not.
+   *
+   * @param name either a fully qualified name or a simple entity name
+   */
+  override def sanitize(name: String)(implicit wp: WskProps): RestResult = {
+    delete(name, DONTCARE_EXIT)
+  }
+}
+
+trait HasActivationRest extends HasActivation {
+
+  /**
+   * Extracts activation id from invoke (action or trigger) or activation get
+   */
+  override def extractActivationId(result: RunResult): Option[String] = {
+    extractActivationIdFromInvoke(result.asInstanceOf[RestResult])
+  }
+
+  /**
+   * Extracts activation id from 'wsk action invoke' or 'wsk trigger invoke'
+   */
+  private def extractActivationIdFromInvoke(result: RestResult): Option[String] = {
+    Try {
+      var activationID =
+        if ((result.statusCode == OK) || (result.statusCode == Accepted)) result.getField("activationId") else ""
+      activationID
```

Comment:
This line isn't needed activationID should already be the last thing returned when it is defined.

Reply:
If I remove activationID, there is going to be an syntax error.
Well, I can change activationID from var to val.